ZEP000-wf-stream-processing.md

```diff
@@ -0,0 +1,1157 @@
+* the `Variable:Created` events create variables for output mapping
+* the `IntermediateCatchEvent:Element_Completed` event transitions the service task element to completed
+* the `Sequence_Flow_Taken` event marks the sequence flow as taken
+
+Differences to current processing:
+* the command `WorkflowInstanceSubscription:Correlate` completes the element directly without writting an `Event_Occurred` event for the element in between 
+
+Considered alternatives:
+* the message variables could be stored as `TemporaryVariable:Created` events before the output mapping is applied
+
+TODO: discuss the message subscription handling. On replay, the subscription must be opened again if it is not opened yet. To open the subscription, the message correlation key expression must be evaluated again. Seems to be too much logic for the replay.
+
+### Interrupting Timer Boundary Event
+
+Command to Process | Written Events and Follow-up Commands
+---|---
+ServiceTask:Activate_Element | ServiceTask:Element_Activating, ServiceTask:Element_Activated, Timer:Created, Job:Created
+Timer:Trigger | Timer:Triggered, ServiceTask:Terminate_Element
+ServiceTask:Terminate_Element | Element_Terminating, ServiceTask:Element_Terminated, BoundaryEvent:Activate_Element 
+BoundaryEvent:Activate_Element | BoundaryEvent:Element_Activating, BoundaryEvent:Element_Activated, BoundaryEvent:Complete_Element
+
+On processing:
+* the `ServiceTask:Activate_Element` command transitions the service task
+  element to activated, creates and schedules a timer, and creates a job
+* the `Timer:Trigger` command triggers the timer, marks the event subscription on the element as triggered, and removes other open event subscriptions from the element
+* the `ServiceTask:Terminate_Element` transitions the service task element to terminated
+* the `BoundaryEvent:Activate_Element` command transitions the boundary event
+  element to activated
+
+On replay:
+* the `ServiceTask:Element_Activating` event transitions the service task element to activating
+* the `ServiceTask:Element_Activated` event transitions the service task element to activated
+* the `Timer:Created` event creates the timer and schedule it
+* the `Job:Created` event creates a job
+* the `Timer:Triggered` event marks the timer as triggered, marks the event subscription on the element as triggered, and removes other open event subscriptions from the element
+* the `ServiceTask:Element_Terminating` event transitions the service task element to terminating
+* the `ServiceTask:Element_Terminating` event transitions the service task element to terminated
+* the `	BoundaryEvent:Element_Activating` event transitions the boundary event element to activating
+* the `	BoundaryEvent:Element_Activated` event transitions the boundary event element to activated
+
+Differences to current processing:
+* the `ServiceTask:Activate_Element` command creates the timer directly without a `Timer:Create` command
```

Comment:
@saig0 Can you explain why we need to change this now?

Reply:
We don't need to remove it. But it makes things easier because we have just one command and transaction instead of two.